articles/quickstart/native/ios-swift/05-authorization.md

```diff
@@ -17,55 +17,59 @@ budicon: 500
+function (user, context, callback) {
 
+  //Define the name of the claim. Must look like a url:
+  //Have 'http' or 'https' scheme and a hostname other than
+  //'auth0.com', 'webtask.io' and 'webtask.run'.
+  var claimName = 'https://access.control/roles';
+
+  //Check if the email has the 'admin.com' domain and give the 'admin' role.
+  //Otherwise, keep a default 'user' role.
+  var roles = ['user'];
+  if (user.email && user.email.indexOf('@admin.com') > -1) {
+      roles.push('admin');
+  }
+  //Set the role claim in the id_token
+  context.idToken[claimName] = roles;
+
+  callback(null, user, context);
+}
 ```
-if (user.email.indexOf('@example.com') > -1)
-```
+
+The rule will run every time a user attempts to authenticate. If the user has a valid `email` and the email's domain is `admin.com`, the user will be granted the `user` and `admin` roles. On any other case, the user will be granted a simple `user` role. This new claim is saved in the *id_token* under the `https://access.control/roles` name, which you will have to check on your app.
 
 ::: note
-You can set roles other than `admin` and `user` or customize the rule as needed.
+Feel free to customize the conditions and the roles to grant on each of them. You can define as many roles as your project requires, beyond `admin` and `user`. There are some restrictions on the naming of the claims detailed in the [rules documentation](/rules/#hello-world).
 :::
 
-By default, it says that if a user email contains `@example.com`, that user will be given an `admin` role, otherwise a regular `user` role.
-
 ## Test the Rule
 
+The id_token is a [JWT](/jwt) that holds several claims, like the one with the roles you set previously. By using a JWT decoding library (like [this one](https://github.com/auth0/JWTDecode.swift)) you will be able to obtain the roles and perform access control.
+
 ```swift
-import Auth0
+import JWTDecode
 ```
 
 ```swift
-// SessionManager.swift
-
-Auth0
-    .users(token: idToken)
-    .get(profile.sub, fields: ["app_metadata"], include: true)
-    .start { result in
-        switch result {
-        case .success(let user):
-            guard
-              let appMetadata = user["app_metadata"] as? [String: Any],
-              let roles = appMetadata["roles"] as? [String]
-            else {
-              // Test failed, make sure you've configured your rule properly (check step 1 thoroughly)
-              return
-            }
-            if roles == "admin" {
-                // User has admin access, grant them the power.
-            } else {
-                // Not an admin, deny the user.
-            }
-        case .failure(let error):
-            // Handler error
-        }
-}
+guard
+    let idToken = self.keychain.string(forKey: "id_token"),
+    let jwt = try? decode(jwt: idToken),
+    let roles = jwt.claim(name: "https://access.control/roles").array
+    else { // Couldn't retrieve claim }
+
+    if roles.contains("admin") {
+        // Authorised
```

Comment:
Does the sample do something when these criteria are met? like allowing the user to access a different screen or something?

Reply:
Yes, I could add that in comments